oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/EventHookCallback.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * EventCallBack: When an alert is present, an event is generated for each alert message. These events are then sent to the internal event analyzer.
+ *
+ */
+public class EventHookCallback implements AlarmCallback {
+
+    private final ModuleManager manager;
+
+    public EventHookCallback(ModuleManager manager) {
+        this.manager = manager;
+    }
+
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessage) {
+        if (null == manager) {
+            return ;
+        }
+        EventAnalyzerService analyzerService = manager.find(EventAnalyzerModule.NAME).provider().getService(EventAnalyzerService.class);
+        alarmMessage.forEach(a -> {
+            analyzerService.analyze(constructCurrentEvent(a));
+        });
+    }
+
+    private Event constructCurrentEvent(AlarmMessage msg) {
+        long millis = System.currentTimeMillis();
+        Event event =  Event.newBuilder()
+                .setUuid(UUID.randomUUID().toString())
+                .setName("Alarm")
+                .setStartTime(millis)
+                .setMessage(msg.getAlarmMessage())
+                .setType(Type.Error)
+                .setEndTime(millis)
+                .build();
+
+        switch (msg.getScopeId()) {
+            case DefaultScopeDefine.SERVICE :
+                IDManager.ServiceID.ServiceIDDefinition serviceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                event = event.toBuilder().setSource(
+                    Source.newBuilder()
+                            .setService(serviceIdDef.getName())
+                            .build()
+                ).build();
+                break;
+            case DefaultScopeDefine.SERVICE_INSTANCE :
+                IDManager.ServiceInstanceID.InstanceIDDefinition instanceIdDef = IDManager.ServiceInstanceID.analysisId(msg.getId0());
+                event = event.toBuilder().setSource(
+                        Source.newBuilder()
+                                .setServiceInstance(instanceIdDef.getName())
+                                .build()
+                ).build();
+                break;
+            case DefaultScopeDefine.ENDPOINT :
+                IDManager.EndpointID.EndpointIDDefinition endpointIDDef = IDManager.EndpointID.analysisId(msg.getId0());
+                event = event.toBuilder().setSource(
+                        Source.newBuilder()
+                                .setEndpoint(endpointIDDef.getEndpointName())
+                                .build()
+                ).build();
+                break;
+            default:
+                event = event.toBuilder().setSource(
+                        Source.newBuilder()
+                                .setService(msg.getName())
+                                .build()
+                ).build();
```

Comment:
Why default should be a service level event?

Reply:
> Why default should be a service level event?

Maybe I should make sure that the scopeId list and the scope info should be set under the corresponding scopeId, which may differ from the three points((Service, Instance, and Endpoint).) on the [UI](https://github.com/apache/skywalking-rocketbot-ui/pull/452#issuecomment-804719314) ， and the default level of event may not be the service .

Reply:
NotifyHandler has covered important logic. UI includes dashboard and topology.

Reply:
Okay,thanks, I'll see what's going on.